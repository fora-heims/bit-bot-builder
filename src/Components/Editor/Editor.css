.editor {
  display: flex;
  gap: 20px;
  justify-content: center;
  align-items: center;
  color: #632a50dd;
}

.selectors,
.selectors * {
  display: flex;
  gap: 20px;
  color: #632a50dd;
  align-items: center;
}

button {
  color: #632a50dd;
  align-items: center;
  background-color: #887fcf33;
  padding: 5px;
  border-radius: 5px;
}

label {
  display: flex;
  flex-direction: column;
}
