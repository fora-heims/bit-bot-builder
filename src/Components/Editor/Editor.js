import './Editor.css';
import React from 'react';

export default function Editor(props) {
  const handleLight = (e) => {
    props.setLight(e.target.value);
    props.setLightCount((prevState) => prevState + 1);
  };
  const handleMid = (e) => {
    props.setMid(e.target.value);
    props.setMidCount((prevState) => prevState + 1);
  };
  const handleDark = (e) => {
    props.setDark(e.target.value);
    props.setDarkCount((prevState) => prevState + 1);
  };
  const handleFeelings = () => {
    props.setFeelings((prevState) => [...prevState, props.feel]);
    props.setFeel('');
  };
  return (
    <div className="editor">
      <div className="selectors">
        <label>
          Light
          <select value={props.light} onChange={handleLight}>
            <option value="light-red">Red</option>
            <option value="light-orange">Orange</option>
            <option value="light-yellow">Yellow</option>
            <option value="light-green">Green</option>
            <option value="light-blue">Blue</option>
            <option value="light-indigo">Indigo</option>
            <option value="light-violet">Violet</option>
          </select>
        </label>
        <label>
          Middle
          <select value={props.mid} onChange={handleMid}>
            <option value="mid-red">Red</option>
            <option value="mid-orange">Orange</option>
            <option value="mid-yellow">Yellow</option>
            <option value="mid-green">Green</option>
            <option value="mid-blue">Blue</option>
            <option value="mid-indigo">Indigo</option>
            <option value="mid-violet">Violet</option>
          </select>
        </label>
        <label>
          Dark
          <select value={props.dark} onChange={handleDark}>
            <option value="dark-redd">Red</option>
            <option value="dark-orange">Orange</option>
            <option value="dark-yellow">Yellow</option>
            <option value="dark-green">Green</option>
            <option value="dark-blue">Blue</option>
            <option value="dark-indigo">Indigo</option>
            <option value="dark-violet">Violet</option>
          </select>
        </label>
      </div>
      <div className="selectors">
        <label>
          What feelings can you identify within yourself while interacting with this image.
          <input type="text" value={props.feel} onChange={(e) => props.setFeel(e.target.value)} />
        </label>
        <button onClick={handleFeelings}>Observe your feelings come and go</button>
      </div>
    </div>
  );
}
