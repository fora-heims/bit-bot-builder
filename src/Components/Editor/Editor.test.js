import { render, screen } from '@testing-library/react';
import Editor from './Editor.js';

test('renders the Display panel', () => {
  const container = render(
    <Editor
      lightCount="5"
      midCount="3"
      darkCount="2"
      feelings={['hello', 'goodbye']}
      feel=""
      light="orange"
      mid="green"
      dark="blue"
    />
  );
  expect(container).toMatchSnapshot();
});
