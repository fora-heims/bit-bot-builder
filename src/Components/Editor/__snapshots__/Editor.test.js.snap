// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Display panel 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="editor"
      >
        <div
          class="selectors"
        >
          <label>
            Light
            <select
              id="light"
            >
              <option
                value="light-redd"
              >
                Red
              </option>
              <option
                value="light-orange"
              >
                Orange
              </option>
              <option
                value="light-yellow"
              >
                Yellow
              </option>
              <option
                value="light-green"
              >
                Green
              </option>
              <option
                value="light-blue"
              >
                Blue
              </option>
              <option
                value="light-indigo"
              >
                Indigo
              </option>
              <option
                value="light-violet"
              >
                Violet
              </option>
            </select>
          </label>
          <label>
            Middle
            <select>
              <option
                value="mid-redd"
              >
                Red
              </option>
              <option
                value="mid-orange"
              >
                Orange
              </option>
              <option
                value="mid-yellow"
              >
                Yellow
              </option>
              <option
                value="mid-green"
              >
                Green
              </option>
              <option
                value="mid-blue"
              >
                Blue
              </option>
              <option
                value="mid-indigo"
              >
                Indigo
              </option>
              <option
                value="mid-violet"
              >
                Violet
              </option>
            </select>
          </label>
          <label>
            Dark
            <select>
              <option
                value="dark-redd"
              >
                Red
              </option>
              <option
                value="dark-orange"
              >
                Orange
              </option>
              <option
                value="dark-yellow"
              >
                Yellow
              </option>
              <option
                value="dark-green"
              >
                Green
              </option>
              <option
                value="dark-blue"
              >
                Blue
              </option>
              <option
                value="dark-indigo"
              >
                Indigo
              </option>
              <option
                value="dark-violet"
              >
                Violet
              </option>
            </select>
          </label>
        </div>
        <div
          class="selectors"
        >
          <label>
            Describe the picture
            <input
              type="text"
              value=""
            />
          </label>
          <button>
            Add description
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="editor"
    >
      <div
        class="selectors"
      >
        <label>
          Light
          <select
            id="light"
          >
            <option
              value="light-redd"
            >
              Red
            </option>
            <option
              value="light-orange"
            >
              Orange
            </option>
            <option
              value="light-yellow"
            >
              Yellow
            </option>
            <option
              value="light-green"
            >
              Green
            </option>
            <option
              value="light-blue"
            >
              Blue
            </option>
            <option
              value="light-indigo"
            >
              Indigo
            </option>
            <option
              value="light-violet"
            >
              Violet
            </option>
          </select>
        </label>
        <label>
          Middle
          <select>
            <option
              value="mid-redd"
            >
              Red
            </option>
            <option
              value="mid-orange"
            >
              Orange
            </option>
            <option
              value="mid-yellow"
            >
              Yellow
            </option>
            <option
              value="mid-green"
            >
              Green
            </option>
            <option
              value="mid-blue"
            >
              Blue
            </option>
            <option
              value="mid-indigo"
            >
              Indigo
            </option>
            <option
              value="mid-violet"
            >
              Violet
            </option>
          </select>
        </label>
        <label>
          Dark
          <select>
            <option
              value="dark-redd"
            >
              Red
            </option>
            <option
              value="dark-orange"
            >
              Orange
            </option>
            <option
              value="dark-yellow"
            >
              Yellow
            </option>
            <option
              value="dark-green"
            >
              Green
            </option>
            <option
              value="dark-blue"
            >
              Blue
            </option>
            <option
              value="dark-indigo"
            >
              Indigo
            </option>
            <option
              value="dark-violet"
            >
              Violet
            </option>
          </select>
        </label>
      </div>
      <div
        class="selectors"
      >
        <label>
          Describe the picture
          <input
            type="text"
            value=""
          />
        </label>
        <button>
          Add description
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
